### PR TITLE
Update template and default paras

### PR DESCRIPTION
**Pull request by QuanWanxx**
1. Update ARM template.
2. Update default parameters in configurations and appsettings.
3. Add deployment doc in **docs** folder.

### CODE REVIEW

File: deploy/templates/FhirSynapseLinkTemplate.json [hunk 1]
```diff
@@ -15,11 +15,15 @@
...
+        "authentication": {
+            "type": "bool",
+            "defaultValue": false,
+            "allowedValues": [
+            true,
+            false
+            ],
             "metadata": {
-                "description": "The name of the Azure API for FHIR resource. Used for managed identity authentication. Leave it empty if you are using an OSS server instance."
+                "description": "Whether fhir server enable authencation checkness, if true will set up managed identity authentication. set false when using an OSS public fhir server instance."
```
Comment (moria97) [inline deploy/templates/FhirSynapseLinkTemplate.json:26]: maybe: `Whether to access Azure API for FHIR with managed identity authentication. Set false if you are using an OSS public FHIR server instance.`

**Review by moria97 — APPROVED** [review]

**Review by BoyaWu10 — APPROVED** [review]

File: deploy/templates/FhirSynapseLinkTemplate.json (hunk 1 above)
Comment (QuanWanxx) [reply deploy/templates/FhirSynapseLinkTemplate.json]: Update in updated FhirSynapsePipelineTemplate.json

File: docs/Deployment.md [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+
+Parameters:
+- **App Name**: azure function name.
+- **Fhir Server Url**: fhir service endpoint. Managed FHIR service is recommended. You can try test with your PaaS fhir server.
+- **Authentication**: whether to access Azure API for FHIR with managed identity authentication. Set false if you are using an OSS public FHIR server instance.
+- **Fhir version**: currently R4 is supported.
+- **Data Start**: start time stamp to export the data.
+- **Data End**: End timestamp to export the data, leave it empty if you want to periodically export data in real time.
+- **Container name**: container in azure storage to store results. The Azure storage will be created by the arm template. You can find the output after deploy completes.
```
Comment (BoyaWu10) [inline docs/Deployment.md:18]: Nit: Capitalize ARM, since it is an abbreviation.

**Review by qiwjin — APPROVED** [review]

File: docs/Deployment.md [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+- **Package url**: the build package of the agent.
+- **App Insight Location**: you can find logs in the deployed application insight resource.
+
+### Test
+The agent will run automatically after deployment. You can find the data written to the target container on azure storage.
+![blob result](./assets/Exported_Data.png)
+
+Now you can try run our script to set up your synapse workspace and explore the data.
```
[the remark this answers is not in the document]
Comment (QuanWanxx) [reply docs/Deployment.md]: Sure, added link to PowerShell script.

File: docs/Deployment.md [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+- **Fhir version**: currently R4 is supported.
+- **Data Start**: start time stamp to export the data.
+- **Data End**: End timestamp to export the data, leave it empty if you want to periodically export data in real time.
+- **Container name**: container in azure storage to store results. The Azure storage will be created by the arm template. You can find the output after deploy completes.
+- **Package url**: the build package of the agent.
+- **App Insight Location**: you can find logs in the deployed application insight resource.
+
+### Test
+The agent will run automatically after deployment. You can find the data written to the target container on azure storage.
+![blob result](./assets/Exported_Data.png)
```
[the remark this answers is not in the document]
Comment (QuanWanxx) [reply docs/Deployment.md]: Sure, updated to ExportedData.png and TemplateParameters.png.

File: docs/Deployment.md [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+### Deployment
+1. Go to [Custom deployment](https://ms.portal.azure.com/#create/Microsoft.Template) on Azure, them select **Build your own template in the editor**.
+2. Use provided [template](../deploy/templates/FhirSynapsePipelineTemplate.json) to deploy an analytics pipeline.
+
+![image](./assets/template_parameters.png)
+
+
+Parameters:
+- **App Name**: azure function name.
+- **Fhir Server Url**: fhir service endpoint. Managed FHIR service is recommended. You can try test with your PaaS fhir server.
```
[the remark this answers is not in the document]
Comment (QuanWanxx) [reply docs/Deployment.md]: Removed this sentence.

File: docs/Deployment.md (hunk 2 above)
Comment (QuanWanxx) [reply docs/Deployment.md]: Sure, updated.

**Pull request merged**